tests: test_is_deletable fails / ResourceWarnings

I am seeing 10 test failures (for `test_is_deletable`) - are they related to recent changes in stripe-mock (I just installed the master version as per the README):

```
======================================================================
FAIL: test_is_deletable (tests.api_resources.test_account.AccountExternalAccountsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_account.py", line 193, in test_is_deletable
    self.assertIsInstance(resource, stripe.BankAccount)
AssertionError: <StripeObject id=this_is_a_fake_id_for_deleted_object at 0x7fe455f47e60> JSON: {
  "deleted": true,
  "id": "this_is_a_fake_id_for_deleted_object"
} is not an instance of <class 'stripe.api_resources.bank_account.BankAccount'>

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_account.AccountTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_account.py", line 95, in test_is_deletable
    '/v1/accounts/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/accounts/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe455dfeb70>, 'delete', '/v1/accounts/acct_1BoJ2AKrZ43eBVAb', {}, None)

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_product.ProductTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_product.py", line 64, in test_is_deletable
    '/v1/products/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/products/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe45588e470>, 'delete', '/v1/products/prod_CCiTUrJyglMaGj', {}, None)

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_customer.CustomerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_customer.py", line 65, in test_is_deletable
    '/v1/customers/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/customers/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe455ac0630>, 'delete', '/v1/customers/cus_CCiTI4Tpghl0nK', {}, None)

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_coupon.CouponTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_coupon.py", line 66, in test_is_deletable
    '/v1/coupons/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/coupons/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe4556da9b0>, 'delete', '/v1/coupons/25OFF', {}, None)

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_apple_pay_domain.ApplePayDomainTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_apple_pay_domain.py", line 43, in test_is_deletable
    '/v1/apple_pay/domains/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/apple_pay/domains/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe4588d8438>, 'delete', '/v1/apple_pay/domains/apwc_1BoJ2MKrZ43eBVAbwqOtiDAK', {}, None)

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_sku.SKUTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_sku.py", line 69, in test_is_deletable
    '/v1/skus/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/skus/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe45582bb38>, 'delete', '/v1/skus/sku_CCiTarxGpuFjAr', {}, None)

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_plan.PlanTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_plan.py", line 67, in test_is_deletable
    '/v1/plans/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/plans/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe45571db00>, 'delete', '/v1/plans/gold', {}, None)

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_recipient.RecipientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_recipient.py", line 64, in test_is_deletable
    '/v1/recipients/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/recipients/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe455ce7e80>, 'delete', '/v1/recipients/rp_1BoJ2NKrZ43eBVAbOlxIQK1Q', {}, None)

======================================================================
FAIL: test_is_deletable (tests.api_resources.test_subscription_item.SubscriptionItemTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "…/Vcs/stripe-python/tests/api_resources/test_subscription_item.py", line 71, in test_is_deletable
    '/v1/subscription_items/%s' % resource.id
  File "…/Vcs/stripe-python/tests/helper.py", line 68, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "…/Vcs/stripe-python/tests/request_mock.py", line 83, in assert_requested
    raise exception
  File "…/Vcs/stripe-python/tests/request_mock.py", line 75, in assert_requested
    self.request_spy.assert_called_with(*args)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 316, in assert_called_with
    return mock.assert_called_with(*args, **kwargs)
  File "…/Vcs/stripe-python/.tox/py36/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: request(<ANY>, 'delete', '/v1/subscription_items/this_is_a_fake_id_for_deleted_object', <ANY>, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x7fe455844b70>, 'delete', '/v1/subscription_items/si_CCiTeY6cvtsSog', {}, None)

----------------------------------------------------------------------
Ran 343 tests in 2.465s

FAILED (failures=10)
Test failed: <unittest.runner.TextTestResult run=343 errors=0 failures=10>
error: Test failed: <unittest.runner.TextTestResult run=343 errors=0 failures=10>
sys:1: ResourceWarning: unclosed <socket.socket fd=47, family=AddressFamily.AF_INET6, type=2049, proto=6, laddr=('::1', 43170, 0, 0), raddr=('::1', 12111, 0, 0)>
…
sys:1: ResourceWarning: unclosed <socket.socket fd=87, family=AddressFamily.AF_INET6, type=2049, proto=6, laddr=('::1', 43542, 0, 0), raddr=('::1', 12111, 0, 0)>
ERROR: InvocationError: '…/Vcs/stripe-python/.tox/py36/bin/python -W all -bb -W error::BytesWarning -m coverage.__main__ run setup.py test'
__________________________________________________________________________________________ summary ___________________________________________________________________________________________
ERROR:   py36: commands failed
  9.06s user 0.76s system 96% cpu 10.189 total
```

There are also a bunch of ResourceWarnings that should get silenced/fixed probably.

--- a/.travis.yml
+++ b/.travis.yml
@@ -22,7 +22,7 @@ cache:
 env:
   global:
     - PYCURL_SSL_LIBRARY=gnutls
-    - STRIPE_MOCK_VERSION=0.8.0
+    - STRIPE_MOCK_VERSION=0.9.0
 
 addons:
   apt:

--- a/tests/api_resources/test_account.py
+++ b/tests/api_resources/test_account.py
@@ -89,12 +89,15 @@ class AccountTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.Account.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/accounts/%s' % resource.id
+            '/v1/accounts/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.Account)
+        self.assertTrue(resource.deleted)
 
     def test_can_retrieve_no_id(self):
         resource = stripe.Account.retrieve()
@@ -190,7 +193,7 @@ class AccountExternalAccountsTests(StripeTestCase):
             '/v1/accounts/%s/external_accounts/%s' % (TEST_RESOURCE_ID,
                                                       TEST_EXTERNALACCOUNT_ID)
         )
-        self.assertIsInstance(resource, stripe.BankAccount)
+        self.assertTrue(resource.deleted)
 
 
 class AccountLoginLinksTests(StripeTestCase):

--- a/tests/api_resources/test_apple_pay_domain.py
+++ b/tests/api_resources/test_apple_pay_domain.py
@@ -37,9 +37,12 @@ class ApplePayDomainTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.ApplePayDomain.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/apple_pay/domains/%s' % resource.id
+            '/v1/apple_pay/domains/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.ApplePayDomain)
+        self.assertTrue(resource.deleted)

--- a/tests/api_resources/test_bank_account.py
+++ b/tests/api_resources/test_bank_account.py
@@ -61,6 +61,8 @@ class BankAccountTest(StripeTestCase):
             'delete',
             '/v1/customers/cus_123/sources/%s' % TEST_RESOURCE_ID
         )
+        # stripe-mock does not yet correctly handle deleting customer sources
+        # self.assertTrue(resource.deleted)
 
     def test_is_verifiable(self):
         resource = self.construct_resource(customer='cus_123')

--- a/tests/api_resources/test_coupon.py
+++ b/tests/api_resources/test_coupon.py
@@ -60,9 +60,12 @@ class CouponTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.Coupon.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/coupons/%s' % resource.id
+            '/v1/coupons/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.Coupon)
+        self.assertTrue(resource.deleted)

--- a/tests/api_resources/test_customer.py
+++ b/tests/api_resources/test_customer.py
@@ -59,12 +59,15 @@ class CustomerTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.Customer.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/customers/%s' % resource.id
+            '/v1/customers/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.Customer)
+        self.assertTrue(resource.deleted)
 
 
 # stripe-mock does not handle the legacy subscription endpoint so we stub

--- a/tests/api_resources/test_plan.py
+++ b/tests/api_resources/test_plan.py
@@ -61,9 +61,12 @@ class PlanTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.Plan.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/plans/%s' % resource.id
+            '/v1/plans/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.Plan)
+        self.assertTrue(resource.deleted)

--- a/tests/api_resources/test_product.py
+++ b/tests/api_resources/test_product.py
@@ -58,9 +58,12 @@ class ProductTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.Product.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/products/%s' % resource.id
+            '/v1/products/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.Product)
+        self.assertTrue(resource.deleted)

--- a/tests/api_resources/test_recipient.py
+++ b/tests/api_resources/test_recipient.py
@@ -58,12 +58,15 @@ class RecipientTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.Recipient.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/recipients/%s' % resource.id
+            '/v1/recipients/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.Recipient)
+        self.assertTrue(resource.deleted)
 
     def test_can_list_transfers(self):
         recipient = stripe.Recipient.retrieve(TEST_RESOURCE_ID)

--- a/tests/api_resources/test_sku.py
+++ b/tests/api_resources/test_sku.py
@@ -63,9 +63,12 @@ class SKUTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.SKU.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/skus/%s' % resource.id
+            '/v1/skus/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.SKU)
+        self.assertTrue(resource.deleted)

--- a/tests/api_resources/test_subscription_item.py
+++ b/tests/api_resources/test_subscription_item.py
@@ -65,9 +65,12 @@ class SubscriptionItemTest(StripeTestCase):
 
     def test_is_deletable(self):
         resource = stripe.SubscriptionItem.retrieve(TEST_RESOURCE_ID)
+        # Unfortunately stripe-mock will return a resource with a different
+        # ID, so we need to store the original ID for the request assertion
+        resource_id = resource.id
         resource.delete()
         self.assert_requested(
             'delete',
-            '/v1/subscription_items/%s' % resource.id
+            '/v1/subscription_items/%s' % resource_id
         )
-        self.assertIsInstance(resource, stripe.SubscriptionItem)
+        self.assertTrue(resource.deleted)

--- a/tests/helper.py
+++ b/tests/helper.py
@@ -13,7 +13,7 @@ from stripe.six.moves.urllib.error import HTTPError
 from tests.request_mock import RequestMock
 
 
-MOCK_MINIMUM_VERSION = '0.5.0'
+MOCK_MINIMUM_VERSION = '0.9.0'
 MOCK_PORT = os.environ.get('STRIPE_MOCK_PORT', 12111)
 
 

